satpy/readers/yaml_reader.py

```diff
@@ -337,7 +337,127 @@ def _build_id_permutations(self, dataset, id_keys):
         return id_kwargs
 
 
-class FileYAMLReader(AbstractYAMLReader, DataDownloadMixin):
+class GenericYAMLReader(AbstractYAMLReader):
+    """A Generic YAML-based reader."""
+
+    def __init__(self, config_dict, filter_parameters=None, filter_filenames=True):
+        """Set up the yaml reader."""
+        super().__init__(config_dict)
+        self.filter_parameters = filter_parameters or {}
+        self.filter_filenames = self.info.get("filter_filenames", filter_filenames)
+
+
+
```

Comment:
```suggestion
```

Reply:
I'm not sure this suggestion is correct, but there should be 2 blank lines less.